var/spack/repos/builtin/packages/proj/package.py

```diff
@@ -144,14 +156,18 @@ def configure_args(self):
             args.append("--with-external-gtest")
 
         if self.spec.satisfies("@7:"):
-            if "+tiff" in self.spec:
-                args.append("--enable-tiff")
-            else:
-                args.append("--disable-tiff")
+            args.extend(self.enable_or_disable("tiff"))
 
             if "+curl" in self.spec:
                 args.append("--with-curl=" + self.spec["curl"].prefix.bin.join("curl-config"))
             else:
                 args.append("--without-curl")
 
+        args.extend(self.enable_or_disable("shared"))
+        args.extend(self.with_or_without("pic"))
+
+        if self.spec["libtiff"].satisfies("+jpeg~shared"):
```

Comment:
This key only exists in the spec when `@7:+tiff`

Reply:
Ok, see if the new logic makes sense